packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -308,6 +331,59 @@ export class IndexedDbPersistence implements Persistence {
     return metadataStore.delete(this.clientId);
   }
 
+  /**
+   * If the garbage collection threshold has passed, prunes the
+   * RemoteDocumentChanges store based on the metadata state of all existing
+   * clients.
+   */
+  // TODO(gsoltis): Execute this as part of LRU GC
+  private maybeGarbageCollectRemoteDocumentChangelog(
+    txn: SimpleDbTransaction
+  ): PersistencePromise<void> {
+    assert(this.isPrimary, 'GC should only run in the primary client');
+
+    if (
+      !this.isWithinAge(
+        this.lastGarbageCollectionTime,
+        CLIENT_METADATA_GARBAGE_COLLECTION_THRESHOLD_MS
+      )
+    ) {
+      this.lastGarbageCollectionTime = Date.now();
+
+      return clientMetadataStore(txn)
+        .loadAll()
+        .next(existingClients => {
+          let activeClients = this.filterActiveClients(
+            existingClients,
+            CLIENT_METADATA_GARBAGE_COLLECTION_THRESHOLD_MS
+          );
+
+          // The primary client doesn't read from the document change log,
+          // and hence we exclude it when we determine the minimum
+          // `lastProcessedDocumentChangesId`.
+          activeClients = activeClients.filter(
+            client => client.clientId !== this.clientId
+          );
+
+          if (activeClients.length) {
+            const mostOutdatedClient = activeClients.reduce(
+              (previousValue, currentValue) =>
+                (currentValue.lastProcessedDocumentChangesId || 0) <
+                (previousValue.lastProcessedDocumentChangesId || 0)
+                  ? currentValue
+                  : previousValue
+            );
+            return this.remoteDocumentCache.removeDocumentChangesThroughChangeId(
+              new IndexedDbTransaction(txn),
+              mostOutdatedClient.lastProcessedDocumentChangesId
```

Comment:
Since we don't actually need `mostOutdatedClient` we can probably simplify all of this a bit, along the lines of:

```
const clientChangeIds = activeClients.map(client => client.lastProcessedDocumentChangesId || 0);
const oldestChangeId = Math.min(...clientChangeIds);
```

Reply:
... but then I can't use `reduce`. It is much simpler, however..